tests/backup/backup_partial_test.go

```diff
@@ -991,3 +992,237 @@ var _ = Describe("{BackupStateTransitionForScheduledBackups}", Label(TestCaseLab
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of partial backup while DE CR for CSI volumes are bring deleted [%s]", backupName))
+			}()
+			wg.Wait()
+		})
+
+		Step("Downgrade the stork version on the destination cluster", func() {
+			log.InfoD("switching to destination context")
+			err := SetDestinationKubeConfig()
+			log.FailOnError(err, "failed to SetClusterContext to destination cluster")
+			CurrentStorkImage, err = GetStorkImageVersion()
+			log.FailOnError(err, "Getting the current stork version on the destination cluster")
+			log.InfoD("Downgrading the stork version on the destination cluster to version - %s", LowerStorkImage)
+			err = DowngradeStorkVersion(LowerStorkImage)
+			log.FailOnError(err, "Downgrading the stork version on the destination cluster")
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Switching context to source cluster failed")
+		})
+
+		Step("Restoring the backed up namespaces in new namespace in destination cluster ", func() {
+			log.InfoD("Restoring the backed up namespaces in new namespace in destination cluster ")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range backedUpNamespaces {
+				namespaceMapping[namespace] = namespace + RandomString(4)
+			}
+			for _, backupName := range backupNames {
+				restoreName := fmt.Sprintf("%s-%s-%s", "default", backupName, RandomString(4))
+				log.InfoD("Restoring from the [%s] backup with namespaceMapping [%v]", backupName, namespaceMapping)
+				err = CreatePartialRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), DestinationClusterName, BackupOrgID, scheduledAppContexts, failedVolumes)
+				if err != nil {
+					dash.VerifyFatal(strings.Contains(err.Error(), "Partial Backups requires Stork version 24.3.0-0 or above. Please upgrade your stork version"), true, fmt.Sprintf("Failure of restore [%s] with error message - %s", restoreName, err.Error()))
+				} else {
+					log.FailOnError(fmt.Errorf("expected error not found for restore [%s]", restoreName), "Expected error not found")
+				}
+			}
+		})
+
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(make([]*scheduler.Context, 0))
+		defer func() {
+			log.Infof("Unsetting BACKUP_TYPE env variable")
+			err := os.Unsetenv("BACKUP_TYPE")
+			log.FailOnError(err, "Unsetting BACKUP_TYPE env variable")
+		}()
+		defer func() {
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+		}()
+		err := SetDestinationKubeConfig()
+		log.FailOnError(err, "Switching context to destination cluster failed")
+		log.Infof("Upgrading stork version on destination cluster to %s ", CurrentStorkImage)
+		err = UpgradeStorkVersion(CurrentStorkImage)
```

Comment:
Can we put this in a defer? As We would want to make sure that the stork image is updated even when the testcase fails?

Reply:
Even if the test case fails, upgrading stork will be the first thing that will be executed. This is why I have added it as a first step.

Reply:
Okay sounds good.